contrib/hash_ring.c

```diff
@@ -172,35 +168,45 @@ int hash_ring_add_items(hash_ring_t *ring, hash_ring_node_t *node) {
     uint64_t keyInt;
 
     // Resize the items array
-    void *resized = realloc(ring->items, (sizeof(hash_ring_item_t*) * ring->numNodes * ring->numReplicas));
+    void *resized = realloc(ring->items, (sizeof(hash_ring_item_t*) * (ring->numItems + node->numReplicas)));
     if(resized == NULL) {
         return HASH_RING_ERR;
     }
     ring->items = (hash_ring_item_t**)resized;
-    for(x = 0; x < ring->numReplicas; x++) {
+
+    uint8_t *data = (uint8_t*)malloc(sizeof(concat_buf) + node->nameLen);
```

Comment:
i've decided to just reuse this buffer instead of allocating it in each iteration of the loop.